### PR TITLE
test: verifreg: Test notaries cannot remove datacap from verifreg

### PR DESCRIPTION
**Pull request by geoff-vball**
## Proposed Changes
Tests the fix implemented [here](https://github.com/filecoin-project/builtin-actors/pull/784). Verifreg notaries should not be able to remove datacap from the verifreg actor, because the verifreg's datacap balance should always equal the datacap of all pending allocations.

## Checklist

Before you mark the PR ready for review, please make sure that:

- [ ] Commits have a clear commit message.
- [ ] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

File: itests/remove_verifreg_datacap_test.go [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+		Params: params,
+		Value:  big.Zero(),
+	}, nil)
+	require.Error(t, err)
+
+	verifregDatacapAfter, err := clientApi.StateVerifiedClientStatus(ctx, builtin.VerifiedRegistryActorAddr, types.EmptyTSK)
+	require.NoError(t, err)
+	require.Equal(t, *verifregDcapBefore, *verifregDatacapAfter) // Verifreg should not have lost datacap
+
+	minerInfo, err := testClient.StateMinerInfo(ctx, testMiner.ActorAddr, types.EmptyTSK)
```
Comment (arajasek) [inline itests/remove_verifreg_datacap_test.go:289]: Are we gaining much by activating this deal? I'm not sure what assertion we're trying to make.

Comment (geoff-vball) [reply itests/remove_verifreg_datacap_test.go:289]: I guess it's a little redundant after checking verifreg's datacap.

**Comment by geoff-vball** [comment]
@arajasek Plz check that the most recent commit seems right to you

**Comment by arajasek** [comment]
Yup, i think that's fine!

**Review by arajasek — APPROVED** [review]

File: chain/stmgr/call.go [hunk 2]
```diff
@@ -133,9 +133,7 @@ func (sm *StateManager) Call(ctx context.Context, msg *types.Message, ts *types.
 
 	// TODO: maybe just use the invoker directly?
 	ret, err := vmi.ApplyImplicitMessage(ctx, msg)
-	if err != nil {
-		return nil, xerrors.Errorf("apply message failed: %w", err)
-	}
+	// Don't check for error here, we want to bubble it up, but also return ret as an InvocResult
```
Comment (simlecode) [inline chain/stmgr/call.go:136]: `ret` may be nil？ @geoff-vball

Comment (geoff-vball) [reply chain/stmgr/call.go:136]: @simlecode you're right. I think this PR has the right behaviour https://github.com/filecoin-project/lotus/pull/9582

**Pull request merged**